tests/testing_config/custom_components/test/cover.py

```diff
@@ -71,8 +72,56 @@ class MockCover(MockEntity, CoverEntity):
     @property
     def is_closed(self):
         """Return if the cover is closed or not."""
+        if self.supported_features & SUPPORT_STOP:
+            return self.current_cover_position == 0
+
+        if "state" in self._values:
+            return self._values["state"] == STATE_CLOSED
+        return False
+
+    @property
+    def is_opening(self):
+        """Return if the cover is opening or not."""
+        if self.supported_features & SUPPORT_STOP:
+            if "state" in self._values:
+                return self._values["state"] == STATE_OPENING
+
         return False
 
+    @property
+    def is_closing(self):
+        """Return if the cover is closing or not."""
+        if self.supported_features & SUPPORT_STOP:
+            if "state" in self._values:
+                return self._values["state"] == STATE_CLOSING
+
+        return False
+
+    def open_cover(self, **kwargs) -> None:
+        """Open cover. This usually needs a thread that simulates the state change from OPENING to OPEN."""
+        if self.supported_features & SUPPORT_STOP:
+            self._values["state"] = STATE_OPENING
+        else:
+            self._values["state"] = STATE_OPEN
+
+    def close_cover(self, **kwargs) -> None:
+        """Close cover. This usually needs a thread that simulates the state change from CLOSING to CLOSED."""
+        if self.supported_features & SUPPORT_STOP:
+            self._values["state"] = STATE_CLOSING
+        else:
+            self._values["state"] = STATE_CLOSED
+
+    def stop_cover(self, **kwargs) -> None:
+        """Stop cover. this is used to simulate the state change which is missing in open_cover and close_cover."""
+        if self.is_opening:
+            self._values["state"] = STATE_OPEN
+            self._values["current_cover_position"] = 100
+        elif self.is_closing:
+            self._values["state"] = STATE_CLOSED
+            self._values["current_cover_position"] = 0
+        else:
+            self._values["state"] = STATE_OPEN if self.is_closed else STATE_CLOSED
+
```

Comment:
I'm not a big fan of this, it makes the tests hard to follow because MockCover's `stop` behaves in a surprising way.

Maybe add a new function instead:
```py
def set_cover_state(state, position):
    self._values["state"] = state
    self._values["current_cover_position"] = position
```

Then you can just call this in the tests to for example change the state from `closing` to `closed`.

Reply:
Fully agree, no surprises! Since i want to test the toggle service and the functions that are called from it, we should not change states directly.

It would not be possible to verify the toggle is working correctly.